solidity/contracts/libraries/staking/TopUps.sol

```diff
@@ -0,0 +1,76 @@
+pragma solidity 0.5.17;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/// @notice TokenStaking contract library allowing to perform two-step stake
+/// top-ups for existing delegations.
+/// Top-up is a two-step process: it is initiated with a declared top-up value
+/// and after waiting for at least the initialization period it can be
+/// committed.
+library TopUps {
+    using SafeMath for uint256;
+
+    struct TopUp {
+        uint256 amount;
+        uint256 createdAt;
+    }
+
+    struct Storage {
+        // operator -> TopUp
+        mapping(address => TopUp) topUps;
+    }
+
+    /// @notice Initiates top-up of the given value for tokens delegated to
+    /// the provided operator. If there is an existing top-up still
+    /// initializing, top-up values are summed up and initialization period
+    /// is set to the current block timestamp.
+    /// @param value Top-up value, the number of tokens added to the stake.
+    /// @param operator Operator The operator with existing delegation to which
+    /// the tokens should be added to.
+    function initiate(
+        Storage storage self,
+        uint256 value,
+        address operator
+    ) public {
+        TopUp memory awaiting = self.topUps[operator];
+        self.topUps[operator] = TopUp(awaiting.amount.add(value), now);
+    }
+
+    /// @notice Commits the top-up if it passed the initialization period.
+    /// Tokens are added to the stake once the top-up is committed.
+    /// @param operator Operator The operator with a pending stake top-up.
+    /// @param initializationPeriod Stake initialization period.
+    function commit(
+        Storage storage self,
+        address operator,
+        uint256 initializationPeriod
+    ) public returns (uint256) {
+        TopUp memory topUp = self.topUps[operator];
+        require(topUp.amount > 0, "No top up to commit");
+        require(
+            now > topUp.createdAt.add(initializationPeriod),
```

Comment:
Why `now` instead `block.timestamp`?

Reply:
`now` will work fine as it's an alias for `block.timestamp`.  We're just inconsistent across our codebase.

Reply:
The only reason for using `now` was probably that it was faster to type it, heh. I can change it to `block.timestamp` if that's what we use in the rest of the codebase, no problem.